### PR TITLE
Make sure namespace packages (e.g. virtualenvwrapper) don't break pex

### LINKED ISSUE #337: virtualenvwrapper breaks Pex (Python 3.5.2 on Ubuntu 16.04)
```console
$ virtualenv env
Running virtualenv with interpreter /usr/bin/python2
New python executable in /tmp/env/bin/python2
Also creating executable in /tmp/env/bin/python
Installing setuptools, pkg_resources, pip, wheel...done.
$ . env/bin/activate
$ pip install pex
Collecting pex
  Using cached pex-1.1.18-py2.py3-none-any.whl
Collecting setuptools<31.0,>=5.7 (from pex)
  Using cached setuptools-30.4.0-py2.py3-none-any.whl
Installing collected packages: setuptools, pex
  Found existing installation: setuptools 32.3.1
    Uninstalling setuptools-32.3.1:
      Successfully uninstalled setuptools-32.3.1
Successfully installed pex-1.1.18 setuptools-30.4.0
$ pex -o require3.5 --python-shebang=/usr/bin/env\ python3.5
$ deactivate
$ python3.5 --version
Python 3.5.2
$ ./require3.5 
Traceback (most recent call last):
  File ".bootstrap/_pex/pex.py", line 324, in execute
  File "/usr/lib/python3.5/contextlib.py", line 59, in __enter__
    return next(self.gen)
  File ".bootstrap/_pex/pex.py", line 250, in patch_sys
  File ".bootstrap/_pex/pex.py", line 210, in minimum_sys
  File ".bootstrap/_pex/pex.py", line 146, in minimum_sys_modules
TypeError: '_NamespacePath' object does not support indexing
```

But it works with Python 3.4:

```console
$ python3.4 ./require3.5 
Python 3.4.5 (default, Jul 15 2016, 16:39:07) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> 
```

### PR DESCRIPTION
**Pull request by itamarst**
Fixes #337.

Let me know what else I can do to get this accepted.

### CODE REVIEW

**Comment by itamarst** [comment]
I'll fix the (minor style) error tomorrow.

**Comment by itamarst** [comment]
OK, updated. Thanks for the quick review!

A speedy release would be nice: for server-side deploys this is less of an issue, but for my use case (distributing developer tools) this is a blocking bug. If anyone else is using pex to distribute developer tools their tool will be breaking at random.

File: pex/pex.py [hunk 1]
```diff
@@ -141,6 +141,14 @@ def minimum_sys_modules(cls, site_libs, modules=None):
         new_modules[module_name] = module
         continue
 
+      # Unexpected objects, e.g. namespace packages, should just be dropped:
+      if module.__path__:
+        try:
+          module.__path__[0]
+        except (TypeError, KeyError, IndexError):
+          TRACER.log('Dropping %s' % (module_name,), V=3)
```
Comment (itamarst) [inline pex/pex.py]: Done.

**Comment by kwlzn** [comment]
@itamarst thanks for the PR! this went out in the 1.1.19 release today.

**Pull request merged**